src/callback_registry.cpp

```diff
@@ -16,17 +16,64 @@
   boost::atomic<uint64_t> nextCallbackNum(0);
 #endif
 
-Callback::Callback(Timestamp when, Task func) : when(when), func(func) {
+// ============================================================================
+// BoostFunctionCallback
+// ============================================================================
+
+BoostFunctionCallback::BoostFunctionCallback(Timestamp when, boost::function<void(void)> func) :
+  Callback(when),
+  func(func)
+{
   this->callbackNum = nextCallbackNum++;
 }
 
+Rcpp::RObject BoostFunctionCallback::rRepresentation() const {
+  using namespace Rcpp;
+  ASSERT_MAIN_THREAD()
+
+  return List::create(
+    _["id"]       = callbackNum,
+    _["when"]     = when.diff_secs(Timestamp()),
+    _["callback"] = Rcpp::CharacterVector::create("C/C++ function")
+  );
+}
+
+
+// ============================================================================
+// RcppFunctionCallback
+// ============================================================================
+
+RcppFunctionCallback::RcppFunctionCallback(Timestamp when, Rcpp::Function func) :
+  Callback(when),
+  func(func)
+{
+  ASSERT_MAIN_THREAD()
+  this->callbackNum = nextCallbackNum++;
+}
+
+Rcpp::RObject RcppFunctionCallback::rRepresentation() const {
+  using namespace Rcpp;
+  ASSERT_MAIN_THREAD()
+
+  return List::create(
+    _["id"]       = callbackNum,
+    _["when"]     = when.diff_secs(Timestamp()),
```

Comment:
I don't love the `Timestamp()` here, as it makes comparison between different rRepresentation objects unreliable. Could this either be the absolute time, or at least have the subtrahend be passed as an argument to `rRepresentation()`?

Reply:
Yeah, that's kind of an ugly bit. Getting the absolute time is a bit tricky to do cross-platform and I figured that since the `list_queue()` function isn't exported and is only for debugging, I'd just do it the easy way.